extensions/reg-core-3.3.md

```diff
@@ -0,0 +1,224 @@
+
+If the server requires a verification token, it MUST also reply with the `RPL_REG_VERIFICATION_REQUIRED`
+numeric, which looks like:
+
+| No. | Label                           | Format                                                                                                       |
+| --- | ------------------------------- | ------------------------------------------------------------------------------------------------------------ |
+| 927 | `RPL_REG_VERIFICATION_REQUIRED` | `:<server> 927 <user_nickname> <accountname> <callback_namespace:><callback> :A verification token was sent` |
+
+Upon error, the IRC server MUST send an error code that is relevant.  We suggest these numerics:
+
+| No. | Label                        | Format                                                                                          |
+| --- | ---------------------------- | ----------------------------------------------------------------------------------------------- |
+| 921 | `ERR_ACCOUNT_ALREADY_EXISTS` | `:<server> 921 <user_nickname> <accountname> :Account already exists`                           |
+| 922 | `ERR_REG_UNSPECIFIED_ERROR`  | `:<server> 922 <user_nickname> <accountname> :<descriptive_text>`                               |
+| 929 | `ERR_REG_INVALID_CALLBACK`   | `:<server> 929 <user_nickname> <accountname> <callback> :Callback token is invalid`             |
+| 928 | `ERR_REG_INVALID_CRED_TYPE`  | `:<server> 928 <user_nickname> <accountname> <cred_type> :Credential type is invalid`           |
+| 440 | `ERR_REG_UNAVAILABLE`        | `:<server> 440 <user_nickname> <subcommand> :Authentication services are currently unavailable` |
+
+The server MAY send additional informative text upon registration success or failure using `RPL_TEXT` or `NOTICE`.
+
+## The `REG VERIFY` sub-command
+
+The `REG VERIFY` command signals the intent of a client to submit a verification token for their
+account to the authentication layer.  The verification token MUST be sent to a valid callback
+resource as specified by the client in the `REG CREATE` command.
+
+A `REG VERIFY` command consists of the following format:
+
+    REG VERIFY <accountname> <auth_code>
+
+The IRC server MAY forward the `REG VERIFY` command to a central authority, or process it
+locally.
+
+Upon success, the IRC server MUST send the `RPL_VERIFYSUCCESS` numeric, which looks like:
+
+| No. | Label               | Format                                                                         |
+| --- | ------------------- | ------------------------------------------------------------------------------ |
+| 923 | `RPL_VERIFYSUCCESS` | `:<server> 923 <user_nickname> <accountname> :Account verification successful` |
+
+The IRC server SHOULD also send an `RPL_LOGGEDIN` (900) numeric and consider the client to be logged in to the
+account that has been successfully verified.
+
+Upon error, the IRC server MUST send an error code that is relevant.  We suggest these
+numerics:
```

Comment:
Could you add the requirement for the error code to be in a given list?
Otherwise, servers may send something a bot does not expect to be a reply to this, and the bot will not know what to do with it.

Reply:
Resolved now